unity/Holo/Assets/Scripts/STLConverter.cs

```diff
@@ -0,0 +1,178 @@
+    }
+
+    // Exports finished GameObject to a .prefab
+    private void ExportToPrefab()
+    {
+        //string save_path = EditorUtility.SaveFilePanel("Choose a folder where for a .prefab file", "", "", ".prefab");
+        string save_path = @"C: \Users\mit - kuchnowski\repos\holo\unity\Holo\Assets\Resources\test2.prefab";
+        PrefabUtility.SaveAsPrefabAsset(series_GameObject, save_path);
+    }
+
+}
+
+//A class for importing a ASTL series from a dir
+public class STLSeriesImporter
+{
+    GameObject imported_STLs = new GameObject("Hypertrophy");
+
+    private string[] file_paths;
+
+    public GameObject Get_GameObject()
+    {
+        return this.imported_STLs;
+    }
+
+    // Object constructor, initiates STL series import.
+    public STLSeriesImporter()
+    {
+        this.Get_file_paths();
+        this.Load_files();
+    }
+
+
+    // gets path to the subsequent STL meshes stored in a root folder.
+    private void Get_file_paths()
+    {
+        //string root_folder = EditorUtility.OpenFolderPanel("Select STL series root folder", "", "");
+        string root_folder = @"C:\Users\mit-kuchnowski\repos\holo\data\hypertrophy-heart\meshes\";
+        file_paths = Directory.GetFiles(root_folder);
+    }
+
+
+    //loads meshes from separate files into one GameObject
+    private void Load_files()
+    {
+        SkinnedMeshRenderer skinnedMesh = imported_STLs.AddComponent<SkinnedMeshRenderer>();
+        Mesh mesh = new Mesh();
+
+        STLFileImporter stlFileImporter = new STLFileImporter();
+
+        bool first_mesh = true;
+        Debug.Log("Doing mesh #1!");
+        stlFileImporter.Load_STL_file(file_paths[0], first_mesh);
+        mesh.vertices = stlFileImporter.BaseVertices;
+        mesh.triangles = stlFileImporter.Indices;
+        mesh.AddBlendShapeFrame(Path.GetFileName(file_paths[0]), 100f, stlFileImporter.AllVertices, null, null);
+        first_mesh = false;
+
+        int count = 2;
+        for (int i = 1; i < file_paths.Length; i++)
+        {
+            Debug.Log("Doing mesh #" + count.ToString() + "!");
+            stlFileImporter.Load_STL_file(file_paths[1], first_mesh);
+            mesh.AddBlendShapeFrame(Path.GetFileName(file_paths[i]), 100f, stlFileImporter.AllVertices, null, null);
+            first_mesh = false;
+            count++;
+        }
+
+        skinnedMesh.sharedMesh = mesh;
+    }
+}
+
+
+//Loads a single STL file and turns it into a list of vertices (x,y,z) & if first_mesh: a list of indexes
+public class STLFileImporter
+{
+    public Vector3[] BaseVertices { get; private set; }
+
+    private List<Vector3> allVertices = new List<Vector3>();
+    public Vector3[] AllVertices { get => allVertices.ToArray(); }
+
+    private List<int> indices = new List<int>();
+    public int[] Indices { get => indices.ToArray(); }
+
+    private uint facetCount = 1;
+
+    public void Load_STL_file(string file_path, bool first_mesh)
+    {
+        allVertices = new List<Vector3>();
```

Comment:
At this point, also recreate `indices`.

Since `AddVertix` updates both `indices` and `allVertices`, then each `Load_STL_file` should clear them both at the beginning.